src/schema.rs

```diff
@@ -287,9 +317,27 @@ impl Schema {
         }
     }
 
+    fn parse_named(complex: &Map<String, Value>, context: &mut SchemaParseContext) -> Result<Self, Error> {
+        let name = Name::parse(complex)?;
+
+        let schema = match complex.get("type") {
+            Some(&Value::String(ref t)) => match t.as_str() {
+                "record" => Schema::parse_record(complex, context),
+                "enum" => Schema::parse_enum(complex),
+                "fixed" => Schema::parse_fixed(complex),
+                _ => panic!("parse_named got wrong type"),
+            },
+            _ => panic!("parse_named got wrong type"),
+        };
+        if schema.is_ok() {
```

Comment:
Would it be possible to have something like:

```rust
if let Ok(ref schema) = schema {
    context.register_type(name, schema.clone());
}
```

? (not sure the borrow checker would be happy, though)